battle-service/src/main/java/com/company/battle/controllers/GameController.java

```diff
@@ -1,43 +1,74 @@
 package com.company.battle.controllers;
 
+import com.company.battle.mappers.GameMapper;
+import com.company.battle.mappers.SettingsMapper;
+import com.company.battle.services.GameService;
 import com.company.common.dtos.GameDto;
-import com.company.common.dtos.SettingsDto;
+import com.company.common.dtos.SearchGameRequestDto;
+import com.company.common.models.GameEntity;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @RestController
 @RequestMapping("api/v1/game")
+@RequiredArgsConstructor
 public class GameController {
 
+    @Autowired
+    private final GameService gameService;
+
+    private final GameMapper gameMapper = GameMapper.INSTANCE;
+
+    private final SettingsMapper settingsMapper = SettingsMapper.INSTANCE;
+
     @GetMapping("/{id}")
     public GameDto getGameById(@PathVariable UUID id) {
-        return null;
+        GameEntity gameEntity = gameService.getById(id);
+        return gameMapper.toDTO(gameEntity);
     }
 
-    @GetMapping("/{link}")
-    public GameDto getGameByLink(@PathVariable String link) {
-        return null;
+    @GetMapping("/public-games")
+    public List<GameDto> getPublicGames() {
+        List<GameEntity> gameEntities = gameService.getPublicGames();
+        return gameEntities.stream()
+                .map(gameMapper::toDTO)
+                .collect(Collectors.toList());
     }
 
-    @GetMapping
-    public List<GameDto> getActiveGames() {
-        return null;
+    @GetMapping("/find-game")
+    public List<GameDto> findGames(@Valid @RequestBody SearchGameRequestDto requestDto) {
+        List<GameEntity> gameEntities = gameService.findGame(requestDto);
+        return gameEntities.stream()
+                .map(gameMapper::toDTO)
+                .collect(Collectors.toList());
     }
 
+    //ToDo: change randomUUID to userId from authorization
     @PostMapping("/create")
-    public GameDto createGame(@RequestBody GameDto gameDto) {
-        return null;
+    public GameDto createGame(@Valid @RequestBody GameDto gameDto) {
+        GameEntity gameEntity = gameMapper.toEntity(gameDto);
+        gameEntity = gameService.create(gameEntity, UUID.randomUUID());
+        return gameMapper.toDTO(gameEntity);
     }
 
-    @PutMapping("/join/{link}")
-    public GameDto joinGame(@PathVariable String link) {
-        return null;
+    //ToDo: Allow connection to only 1 game
+    @PutMapping("/join/{id}")
+    public GameDto joinGame(@PathVariable UUID id, @PathVariable UUID userId) {
```

Comment:
`  public GameDto joinGame(@PathVariable UUID id) {`
user id should be exteracted from security context. even if you don't have it, you can use fake value so far